Update .gitignore.

domob1812 opened the pull request:
Update the top-level .gitignore to fit the generated files I see on a
GNU/Linux system more closely.

domob1812 committed:
Update .gitignore.

Update the top-level .gitignore to fit the generated files I see on a
GNU/Linux system more closely.

belovachap commented on .gitignore:
```diff
@@ -1,7 +1,10 @@
 src/namecoind
+src/namecoind.exe
+build
+Makefile
```
Don't we want to track the Makefile?

domob1812 replied:
No, the (top-level) Makefile is created by qmake.

belovachap commented on .gitignore:
```diff
@@ -1,7 +1,10 @@
-src/bitcoind
 .*.swp
 *.*~*
+qrc_bitcoin.cpp
```
What's qrc_bitcoin.cpp and why do we want to ignore it?

belovachap commented on .gitignore:
```diff
@@ -1,7 +1,10 @@
+namecoin-qt.exe
 src/namecoind
+src/namecoind.exe
+build
```
Not familiar with the build process, what command do you run that creates a build directory?

domob1812 commented:
I'm also not familiar with the build system (Qt) very much ... maybe some of these are relicts of older versions.  It would be cool if someone especially on other systems could test further (or with more Qt experience could chim in).  The Makefile is definitely generated by Qt, though (see the top lines), and the namecoin-qt binary should be ignored, too.

belovachap commented:
Ah, I think I see what you're trying to do. A couple of thoughts: 1) Makefile will ignore Makefiles in all directories (probably want `/Makefile` if we just want to ignore the top level one; http://stackoverflow.com/questions/3637660/how-to-exclude-file-only-from-root-folder-in-git). It kind of works as-is because the Makefile in src/ is already being tracked. 2) Gathering together `/qrc_bitcoin.cpp`, `build/`, and `/Makefile` under a comment like `# Qt build crud` might give a good hint about why these are in the .gitignore file.

domob1812 committed:
Update .gitignore further.

Clarify with a comment that the ignored files are Qt-created, and
exclude only toplevel Makefile.

domob1812 commented:
Ok, I've updated the pull request accordingly.

belovachap commented:
Looks good to me! :)

domob1812 committed:
Merge branch 'namecoinq' into gitignore
